# Теория вероятностей и математическая статистика

### Урок 2. Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

Урок 2.

1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.
2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?
3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?
4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

In [1]:
from math import factorial
import pandas as pd
import numpy as np

Число сочетаний

$$C_{n}^k =\frac{n!}{k!(n - k)!}$$ 

In [2]:
def combinations(n, k):
    # Число сочетаний из  𝑛  элементов по  𝑘  элементов в каждом (в сочетаниях порядок не важен):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

## Задача 1

Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

- X - дискретная случайная величина
- p = 0.8
- n = 100
- k = 85

поэтому используем формулу Бернули.
$$P_n(x=k)=C_{n}^k p^k q^{n-k}$$

для нашего случая
$$P_{100}(x=85)=C_{100}^{85} p^{85} q^{15}$$ 

p = 0.8, q= 1 - p = 0.2

In [3]:
n=100
k=85
p=0.8
q=1-p

C_n_k=combinations(n, k)
C_n_k

253338471349988640

In [4]:
P=C_n_k*(p**k)*q**(n-k)

print(f"Вероятность того, что стрелок попадет в цель ровно 85 раз P={P :.3f}")

Вероятность того, что стрелок попадет в цель ровно 85 раз P=0.048


**Ответ**: Вероятность того, что стрелок попадет в цель ровно 85 раз P=0.048

## Задача 2

Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна p=0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек.
- Какова вероятность, что ни одна из них не перегорит в первый день?
- Какова вероятность, что перегорят ровно две?

- X - дискретная случайная величина
- $p = 0.0004 < 0.1$
- $n = 5000$ - достаточно большое
- $m_1=0$ - А - не перегорела ни одна лампочка
- $m_2=2$ - B - перегорело ровно 2 лампочки

Поэтому используем формулу Пуассона.
$$P_m=\frac{\lambda^m}{m!}{e^{-\lambda}}$$

In [5]:
p=0.0004
n=5000

### 2.1 Какова вероятность, что ни одна из них не перегорит в первый день? 

$$m_1=0$$

In [6]:
m=0
L=p*n
print(f"lambda={L}")

lambda=2.0


Учитываем, что 
$$P = 1 - P_m$$

In [7]:
exp=2.72
P_0=1 - ((L**m)/(factorial(m)))*exp**(-L)
print(f"Вероятность, что ни одна из них не перегорит в первый день P(0)={P_0 :.3f}")
P_0

Вероятность, что ни одна из них не перегорит в первый день P(0)=0.865


0.8648356401384083

**Ответ**: Вероятность, что ни одна из них не перегорит в первый день P(0) = 86.5%

## 2.2 Какова вероятность, что перегорят ровно две?
$$m_2=2$$

In [8]:
m=2
L=p*n
print(f"lambda={L}")

lambda=2.0


Учитываем, что 
$$P = 1 - P_m$$

In [9]:
P_2 = 1 - ((L**m)/(factorial(m)))*exp**(-L)
print(f"Вероятность, что перегорят ровно две в первый день P(2)={P_2 :.3f}")
P_2

Вероятность, что перегорят ровно две в первый день P(2)=0.730


0.7296712802768166

**Ответ**: Вероятность, что перегорят ровно две в первый день P(2) = 73%

## Задача 3

Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

- X - дискретная случайная величина
- p = 1/2
- q = 1 - p =1/2
- n = 144
- k = 70

Поэтому используем формулу Бернули:

$$P_n(x=k)=C_{n}^k p^k q^{n-k}$$

$$C_{n}^k =\frac{n!}{k!(n - k)!}$$

In [10]:
n=144
k=70
p=0.5
q=1-p

C_n_k=combinations(n, k)
C_n_k

1400749509083708712353342007350408066891776

In [11]:
P=C_n_k*(p**k)*q**(n-k)

print(f"Вероятность того,  что орел выпадет ровно 70 раз P = {P :.3f}")

Вероятность того,  что орел выпадет ровно 70 раз P = 0.063


**Ответ**: Вероятность того,  что орел выпадет ровно 70 раз P = 6.3%

## Задача 4

В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых.

Из каждого ящика вытаскивают случайным образом по два мяча.
- Какова вероятность того, что все мячи белые?
- Какова вероятность того, что ровно два мяча белые?
- Какова вероятность того, что хотя бы один мяч белый?

### 4.1 Какова вероятность того, что все мячи белые?


А1 - мячи, вынутые из первого ящика все белые

А2 - мячи, вынутые из второго ящика все белые

А = А1 * А2 - все мячи белые

$P(A1) =  \frac{C_{7}^{2}}{C_{10}^{2} }$

$P(A2) =  \frac{C_{9}^{2}}{C_{11}^{2} }$

$P(A) = P(A1)*P(A2)$

In [12]:
P_A1 = combinations(7, 2)/combinations(10, 2)
print(f"P_A1 = {P_A1*100:.1f} %")
P_A2 = combinations(9, 2)/combinations(11, 2)
print(f"P_A2 = {P_A2*100:.1f} %")

P_A1 = 46.7 %
P_A2 = 65.5 %


In [13]:
P_A = P_A1 * P_A2
print(f"P_A = {P_A*100:.1f}%")

P_A = 30.5%


**Ответ**: Вероятность того, что все мячи белые P(A)=30.5%

### 4.2 Какова вероятность того, что ровно два мяча белые?

A11- из первого ящика вынули два белых мяча

А12 - из второго ящика вынули только черные мячи

А21 - из второго ящика вынули два белых мяча

А22 - из первого ящика вынули только черные мячи

А31 - из первого ящика вынули один белый мячи

А32 - из второго ящика вынули один белый мячи

А = А11 А12 + А21 А22 + А31 * А32 - вынули ровно два белых мяча

$P(A11) =  \frac{C_{7}^{2}}{C_{10}^{2} }$

$P(A12) =\frac{C_{2}^{2}}{C_{11}^{2} }$

In [14]:
P_A11 = combinations(7, 2)/combinations(10, 2)
print(f"P_A11 = {P_A11*100:.1f} %")
P_A12 = combinations(2, 2)/combinations(11, 2)
print(f"P_A2 = {P_A2*100:.1f} %")

P_A11 = 46.7 %
P_A2 = 65.5 %


$P(A21) =  \frac{C_{9}^{2}}{C_{11}^{2} }$

$P(A22) =\frac{C_{3}^{2}}{C_{10}^{2} }$

In [15]:
P_A21 = combinations(9, 2)/combinations(11, 2)
print(f"P_A21 = {P_A21*100:.1f} %")
P_A22 = combinations(3, 2)/combinations(10, 2)
print(f"P_A22 = {P_A22*100:.1f} %")

P_A21 = 65.5 %
P_A22 = 6.7 %


$P(A31) = \frac{C_{7}^{1}}{C_{10}^{1}}*\frac{C_{3}^{1}}{C_{10}^{1}} = \frac{C_{7}^{1} * C_{3}^{1}}{C_{10}^{2}}$

$P(A31) = \frac{C_{9}^{1}}{C_{11}^{1}}*\frac{C_{2}^{1}}{C_{11}^{1}} = \frac{C_{9}^{1} * C_{2}^{1}}{C_{11}^{2}}$

In [16]:
P_A31 = (combinations(7, 1)*combinations(3, 1))/(combinations(10, 2)^2)
print(f"P_A31 = {P_A31*100:.1f} %")
P_A32 = (combinations(9, 1)*combinations(2, 1))/(combinations(11, 2)^2)
print(f"P_A32 = {P_A32*100:.1f} %")

P_A31 = 44.7 %
P_A32 = 34.0 %


In [17]:
PA = P_A11*P_A12 + P_A21*P_A22 + P_A31*P_A32
print(f'Вероятность того, что вынут ровно два белых мяча = {PA :.3f}')

Вероятность того, что вынут ровно два белых мяча = 0.204


**Ответ**: Вероятность того, что вынут ровно два белых мяча P(A)= 20.4%

### 4.3 Какова вероятность того, что хотя бы один мяч белый?

А12 - из второго ящика вынули только черные мячи

А22 - из первого ящика вынули только черные мячи

A = A12A22

P(1) = 1 - P(0) = 1 - P_A12*P_22

In [18]:
P_1 = 1 - P_A12*P_A2
P_1
print(f'Вероятность того, что что хотя бы один мяч белый = {P_1 :.3f}')

Вероятность того, что что хотя бы один мяч белый = 0.988


**Ответ**: Вероятность того, что что хотя бы один мяч белый = 98.8%